test/smoke/framework/testCore/src/main/java/com/microsoft/applicationinsights/smoketest/AiSmokeTest.java

```diff
@@ -376,11 +407,76 @@ protected static void checkMockedIngestionHealth() throws Exception {
 		assertEquals(MockedAppInsightsIngestionServlet.PONG, postResponse);
 	}
 
-	protected static void startDockerContainer() throws Exception {
+	private static void createDockerNetwork() throws Exception {
+		try {
+			System.out.printf("Creating network '%s'...%n", networkName);
+			networkId = docker.createNetwork(networkName);
+		} catch (Exception e) {
+			System.err.printf("Error creating network named '%s'%n", networkName);
+			e.printStackTrace();
+			throw e;
+		}
+	}
+
+	private static void cleanUpDockerNetwork() throws Exception {
+		if (networkId == null) {
+			System.out.println("No network id....nothing to clean up");
+			return;
+		}
+		try {
+			System.out.printf("Deleting network '%s'...%n", networkName);
+			docker.deleteNetwork(networkName);
+		} catch (Exception e) {
+			System.err.printf("Error deleting network named '%s' (%s)%n", networkName, networkId);
+			e.printStackTrace();
+		} finally {
+			networkId = null;
+		}
+	}
+
+	private static void startAllContainers() throws Exception {
+		startDependencyContainers();
+		startTestApplicationContainer();
+	}
+
+	private static void startDependencyContainers() throws IOException, InterruptedException {
+		if (dependencyImages.isEmpty()) {
+			System.out.println("No dependency containers to start.");
+			return;
+		}
+
+		for (DependencyContainer dc : dependencyImages) {
+			String imageName = Strings.isNullOrEmpty(dc.imageName()) ? dc.value() : dc.imageName();
+			System.out.printf("Starting container: %s%n", imageName);
+			final String containerId = docker.startContainer(imageName, dc.portMapping(), networkId);
+			assertFalse("'containerId' was null/empty attempting to start container: "+imageName, Strings.isNullOrEmpty(containerId));
+			System.out.printf("Dependency container started: %s (%s)%n", imageName, containerId);
+
+			String containerName = docker.getRunningContainerName(containerId);
+			if (containerName == null) {
+				String message = String.format("Could not get container name for id=%s. ", containerId);
+				if (docker.isContainerRunning(containerId)) {
+					message += "It appears to be running.";
```

Comment:
Do you explicitly depend on ContainerName? If yes, then this message seems to be bit misleading. Shouldn't the container stop in this case or forced stop?

Ideally I believe either name or id of the container is sufficient correct?

Reply:
This is just checking for inconsistencies to assist in debugging. Both should be defined